fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/HashDistributionDesc.java

```diff
@@ -18,7 +18,8 @@
         //  the result will be error
         SHUFFLE_AGG, // hash property from shuffle agg
         SHUFFLE_JOIN, // hash property from shuffle join
-        BUCKET_JOIN // hash property from bucket join
+        BUCKET_JOIN, // hash property from bucket join
+        SHUFFLE_ENFORCE // parent node which can not satisfy the requirement will enforce child this hash property
```

Comment:
why we need to know the hash property whether is enforce？

Reply:
enforce type enforcer is forced generated，no matter what the child's output property is. we need add this type enforcer when  there colocate join can not satisfied or with hint[SHUFFLE]